### PR TITLE
fix: limit amount by decimal

### PR DESCRIPTION
**Pull request by zhouhanseng**
[MASK-377](https://mask.atlassian.net/jira/software/c/projects/MASK/boards/15?modal=detail&selectedIssue=MASK-377)

### CODE REVIEW

**Comment by zhouhanseng** [comment]
@DimensionDev/qa 

There's a similar issue request for test: https://github.com/DimensionDev/Maskbook/pull/5086

**Comment by zhouhanseng** [comment]
@DimensionDev/qa ready for test

**Comment by orange187** [comment]
Checked

File: packages/mask/src/plugins/ITO/SNSAdaptor/CreateForm.tsx [hunk 1]
```diff
@@ -183,13 +183,22 @@ export function CreateForm(props: CreateFormProps) {
...
 
-    const onTotalOfPerWalletChange = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
-        const total = ev.currentTarget.value
-        if (total === '') setTotalOfPerWallet('')
-        if (/^\d+\.?\d*$/.test(total)) {
-            setTotalOfPerWallet(total)
-        }
-    }, [])
+    const RE_MATCH_WHOLE_AMOUNT = useMemo(
+        () => new RegExp(`^\\d*\\.?\\d{0,${tokenAndAmount?.token?.decimals}}$`), // d.ddd...d
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/CreateForm.tsx:187]:
```suggestion
        () => new RegExp(`^\\d*\\.?\\d{0,${tokenAndAmount?.token?.decimals ?? 0}}$`), // d.ddd...d
```

Comment (zhouhanseng) [reply packages/mask/src/plugins/ITO/SNSAdaptor/CreateForm.tsx:187]: I think fallback to 18 maybe better.

**Review by guanbinrui — APPROVED** [review]

**Review by yanzhihong23 — APPROVED** [review]

**Pull request merged**